### PR TITLE
Drops inventory on disconnect if player is in a match.

### LINKED ISSUE #302: Drop inventory on leave
When players leave the game by any means, their inventory is deleted. The inventory should drop as if they got died.

Reintroduce tag because this was the case on OCN.

### PR DESCRIPTION
**Pull request by KingOfSquares**
Should fix #302 :)

EDIT: Saw later that it should require a tag(in map.xml), so i will work on adding that now.
EDIT2: NVM, i misunderstood. This PR should be ready for review now.

Signed-off-by: KingOfSquares <simonmorland@gmail.com>

### CODE REVIEW

File: src/main/java/tc/oc/pgm/listeners/PGMListener.java [hunk 1]
```diff
@@ -303,5 +305,27 @@ public void nerfFishing(PlayerFishEvent event) {
...
       }
     }
+
+
+  }
+
+  @EventHandler(priority = EventPriority.MONITOR)
+  public void dropItemsOnQuit(PlayerParticipationStopEvent event) {
+    MatchPlayer quitter = event.getPlayer();
+    Player player = Bukkit.getPlayer(quitter.getId());
```
Comment (TheMolkaPL) [inline src/main/java/tc/oc/pgm/listeners/PGMListener.java]: There must be a getter for this in `MatchPlayer` for sure.

File: src/main/java/tc/oc/pgm/listeners/PGMListener.java [hunk 2]
```diff
@@ -303,5 +305,27 @@ public void nerfFishing(PlayerFishEvent event) {
...
+
+      for (ItemStack i : quitter.getInventory().getContents()) { //Drop inventory
```
Comment (TheMolkaPL) [inline src/main/java/tc/oc/pgm/listeners/PGMListener.java]: Can we not call variables with single letters? Can we call this `item` instead?

File: src/main/java/tc/oc/pgm/listeners/PGMListener.java (hunk 1 above)
Comment (KingOfSquares) [reply src/main/java/tc/oc/pgm/listeners/PGMListener.java]: You mean MatchPlayer.getBukkit()? It was deprecated so i avoided it

File: src/main/java/tc/oc/pgm/listeners/PGMListener.java [hunk 3]
```diff
@@ -303,5 +305,27 @@ public void nerfFishing(PlayerFishEvent event) {
...
+        }
+      }
+
+      ItemStack[] playerArmor = quitter.getInventory().getArmorContents(); //Drop Armor
+      for (int i = 0; i < 4 ; i++) {
+        if(playerArmor[i].getType() == Material.AIR){ //Skips empty armor slots
+          continue;
+        }
+        quitter.getWorld().dropItemNaturally(player.getLocation(), playerArmor[i]);
+      }
```
Comment (TheMolkaPL) [inline src/main/java/tc/oc/pgm/listeners/PGMListener.java]: Are you sure about this? I think `getContents()` already returns the whole inventory with the armor.

Comment (KingOfSquares) [reply src/main/java/tc/oc/pgm/listeners/PGMListener.java]: No, just tested it again to be sure. When i remove this part of the code it only drops items from the Inventory.

File: src/main/java/tc/oc/pgm/listeners/PGMListener.java (hunk 2 above)
Comment (KingOfSquares) [reply src/main/java/tc/oc/pgm/listeners/PGMListener.java]: Good idea! Will change boths "i"s to "item"

File: src/main/java/tc/oc/pgm/listeners/PGMListener.java (hunk 1 above)
Comment (Electroid) [reply src/main/java/tc/oc/pgm/listeners/PGMListener.java]: @KingOfSquares No, please use `MatchPlayer#getBukkit`

File: src/main/java/tc/oc/pgm/listeners/PGMListener.java [hunk 4]
```diff
@@ -304,4 +306,24 @@ public void nerfFishing(PlayerFishEvent event) {
       }
     }
   }
+
+  @EventHandler(priority = EventPriority.MONITOR)
+  public void dropItemsOnQuit(PlayerParticipationStopEvent event) {
+    MatchPlayer quitter = event.getPlayer();
+    Player player = Bukkit.getPlayer(quitter.getId());
+
+      for (ItemStack item : quitter.getInventory().getContents()) { //Drop inventory
```
Comment (Electroid) [inline src/main/java/tc/oc/pgm/listeners/PGMListener.java]: No need for the `//` comments in this method, it's self explanatory.

You really only need comments for things that might be harder to understand.

File: src/main/java/tc/oc/pgm/listeners/PGMListener.java [hunk 5]
```diff
@@ -304,4 +306,24 @@ public void nerfFishing(PlayerFishEvent event) {
...
+        if (item != null) {
+          quitter.getWorld().dropItemNaturally(player.getLocation(), item);
+        }
+      }
+
+      ItemStack[] playerArmor = quitter.getInventory().getArmorContents(); //Drop Armor
```
Comment (Electroid) [inline src/main/java/tc/oc/pgm/listeners/PGMListener.java]:
```java
for (ItemStack armor : quitter.getInventory().getArmorContents()) {
  if(armor == null || armor == Material.AIR) continue;
  quitter.getWorld().dropItemNaturally(player.getLocation(), armor);
}

File: src/main/java/tc/oc/pgm/listeners/PGMListener.java [hunk 6]
```diff
@@ -304,4 +306,24 @@ public void nerfFishing(PlayerFishEvent event) {
...
     }
   }
+
+  @EventHandler(priority = EventPriority.MONITOR)
+  public void dropItemsOnQuit(PlayerParticipationStopEvent event) {
+    MatchPlayer quitter = event.getPlayer();
+    Player player = Bukkit.getPlayer(quitter.getId());
+
+      for (ItemStack item : quitter.getInventory().getContents()) { //Drop inventory
+        if (item != null) {
```
Comment (Electroid) [inline src/main/java/tc/oc/pgm/listeners/PGMListener.java]: You can structure this similar to how I did the armor below. You also need to check that the `item` is not `AIR`.

File: src/main/java/tc/oc/pgm/listeners/PGMListener.java (hunk 4 above)
Comment (KingOfSquares) [reply src/main/java/tc/oc/pgm/listeners/PGMListener.java]: Ok! Will remove

File: src/main/java/tc/oc/pgm/listeners/PGMListener.java (hunk 5 above)
Comment (KingOfSquares) [reply src/main/java/tc/oc/pgm/listeners/PGMListener.java]: Im assuming you meant `armor.getType() == Material.AIR`

File: src/main/java/tc/oc/pgm/listeners/PGMListener.java (hunk 6 above)
Comment (KingOfSquares) [reply src/main/java/tc/oc/pgm/listeners/PGMListener.java]: Ok!

File: src/main/java/tc/oc/pgm/listeners/PGMListener.java [hunk 7]
```diff
@@ -310,20 +310,16 @@ public void nerfFishing(PlayerFishEvent event) {
...
     MatchPlayer quitter = event.getPlayer();
-    Player player = Bukkit.getPlayer(quitter.getId());
 
-      for (ItemStack item : quitter.getInventory().getContents()) { //Drop inventory
-        if (item != null) {
-          quitter.getWorld().dropItemNaturally(player.getLocation(), item);
-        }
+      for (ItemStack item : quitter.getInventory().getContents()) {
+        if (item == null || item.getType() == Material.AIR) continue;
+          quitter.getWorld().dropItemNaturally(quitter.getBukkit().getLocation(), item);
```
Comment (TheMolkaPL) [inline src/main/java/tc/oc/pgm/listeners/PGMListener.java]: Redundant indentation here.

File: src/main/java/tc/oc/pgm/listeners/PGMListener.java [hunk 8]
```diff
@@ -310,20 +310,16 @@ public void nerfFishing(PlayerFishEvent event) {
...
 
-      ItemStack[] playerArmor = quitter.getInventory().getArmorContents(); //Drop Armor
-      for (int item = 0; item < 4 ; item++) {
-        if(playerArmor[item].getType() == Material.AIR){ //Skips empty armor slots
-          continue;
-        }
-        quitter.getWorld().dropItemNaturally(player.getLocation(), playerArmor[item]);
+      for (ItemStack armor : quitter.getInventory().getArmorContents()) {
+        if(armor == null || armor.getType() == Material.AIR) continue;
+          quitter.getWorld().dropItemNaturally(quitter.getBukkit().getLocation(), armor);
```
Comment (TheMolkaPL) [inline src/main/java/tc/oc/pgm/listeners/PGMListener.java]: Redundant indentation here.

File: src/main/java/tc/oc/pgm/listeners/PGMListener.java [hunk 9]
```diff
@@ -7,6 +7,7 @@
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.locks.ReentrantReadWriteLock;
 import net.md_5.bungee.api.ChatColor;
+import org.bukkit.Bukkit;
```
Comment (TheMolkaPL) [inline src/main/java/tc/oc/pgm/listeners/PGMListener.java]: Redundant import here.

Comment (KingOfSquares) [reply src/main/java/tc/oc/pgm/listeners/PGMListener.java]: Yes, old import. Will remove

File: src/main/java/tc/oc/pgm/listeners/PGMListener.java (hunk 7 above)
Comment (KingOfSquares) [reply src/main/java/tc/oc/pgm/listeners/PGMListener.java]: I love indentation too much 😢 , will remove

File: src/main/java/tc/oc/pgm/listeners/PGMListener.java (hunk 8 above)
Comment (KingOfSquares) [reply src/main/java/tc/oc/pgm/listeners/PGMListener.java]: I love indentation too much 😢 , will remove

**Comment by Electroid** [comment]
@BGMP or @KingOfSquares you'll need to fix the conflict and rebase to master, then this thing is ready to go!

**Comment by BGMP** [comment]
@KingOfSquares ;-;

**Comment by KingOfSquares** [comment]
> @KingOfSquares ;-;

I got this 😎

**Pull request closed without merging**